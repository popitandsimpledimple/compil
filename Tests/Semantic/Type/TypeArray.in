type arr = array[0..100] of integer;
begin end.