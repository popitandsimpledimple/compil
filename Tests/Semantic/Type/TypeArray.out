program
   decls
      decls types
         decl type
            arr
               array
                  range
                     ..
                        0
                        100
            type
               integer
   block
      compounds statements

 ───────────────────────────────────────────────────────────────────────────────────────────────────
 │ integer                   │ type                      │ integer                                 │
 │ char                      │ type                      │ char                                    │
 │ string                    │ type                      │ string                                  │
 │ double                    │ type                      │ double                                  │
 │ boolean                   │ type                      │ boolean                                 │
 ───────────────────────────────────────────────────────────────────────────────────────────────────

 ───────────────────────────────────────────────────────────────────────────────────────────────────
 │ without name              │ program name              │ no return type                          │
 │ arr                       │ alias                     │ array                                   │
 ───────────────────────────────────────────────────────────────────────────────────────────────────
